### ISSUE TITLE
[BUG] Marlin-2.0.5 binary a lot bigger than 2.0.4.4 with same config

### ISSUE THREAD

**Issue by Marlor** [MarlinFirmware/Marlin#17172]
### Bug Description

The Marlin-2.0.5 binary is a hefty chunk bigger than in 2.0.4.4 with the same config
Is this a bug or intended?

The only differences in the configs are things like,
the renaming from HAS_TRINAMIC to HAS_TRINAMIC_CONFIG etc.

### My Configurations

[Marlin-2.0.4.4.zip](https://github.com/MarlinFirmware/Marlin/files/4334695/Marlin-2.0.4.4.zip)
firmware.bin 261 KB
```
RAM:   [========  ]  76.9% (used 25184 bytes from 32736 bytes)
Flash: [======    ]  56.1% (used 266380 bytes from 475136 bytes)
```

[Marlin-2.0.5.zip](https://github.com/MarlinFirmware/Marlin/files/4334697/Marlin-2.0.5.zip)
firmware.bin 326 KB
```
RAM:   [=======   ]  69.4% (used 22728 bytes from 32736 bytes)
Flash: [=======   ]  70.2% (used 333452 bytes from 475136 bytes)

```

**Comment by thinkyhead** [comment]
Which feature is the one that —when turned off— shows no significant size difference?

**Comment by ellensp** [comment]
Interesting. the firmware.elf files are about the same. but the bin file generated from the .elf is way larger

**Comment by Marlor** [comment]
The tests i could do today:

**default Marlin Config:**
Marlin-2.0.4.4 
firmware.hex **90 KB**

Marlin-2.0.5
firmware.hex **90 KB**



**default CR10s Config:**
Marlin-2.0.4.4 
firmware.hex **434 KB**

Marlin-2.0.5
firmware.hex **435 KB**



**default Marlin Config only change BOARD_BTT_SKR_V1_3:**
Marlin-2.0.4.4 
firmware.bin **85 KB**

Marlin-2.0.5
firmware.bin **85 KB**



**default CR10s Config only change BOARD_BTT_SKR_V1_3:**
Marlin-2.0.4.4 
firmware.bin **169 KB**

Marlin-2.0.5
firmware.bin **235 KB**


So it has to be a feature which is off in default, on in default CR10s
And the bloat does only happen with a LPC1768 Board (maby more) but not on Atmega

**Comment by thinkyhead** [comment]
I've been testing build sizes, and here's what I can tell you…

With Marlin 2.0.5 if you want to use `FLASH_EEPROM_EMULATION` you have to specify it, otherwise it now falls back to `SDCARD_EEPROM_EMULATION` for LPC1768.

**Comment by Marlor** [comment]
Sounds like an easy fix, add

```
// Use one of these or SDCard-based Emulation will be used
//#define SRAM_EEPROM_EMULATION   // Use BackSRAM-based EEPROM emulation
#define FLASH_EEPROM_EMULATION  // Use Flash-based EEPROM emulation
```
to `pins_BTT_SKR_V1_3.h`

But maybe more boards have slipped through the cracks

**Comment by thinkyhead** [comment]
Sure thing. I didn't want to presume which boards should get which form of emulation by default. It seems most sensible to fall back to SD because this is the least volatile. We don't seem to have an "SRAM" option for LPC1768.

Previously with 2.0.4 you couldn't choose SD-based EEPROM emulation at all because the `persistent_store_api.h` file always defines `FLASH_EEPROM_EMULATION`. So it is now possible to choose, but yep, the fallback is SD.

**Comment by thisiskeithb** [comment]
Edit: I created a separate issue: https://github.com/MarlinFirmware/Marlin/issues/17179

**Comment by thinkyhead** [comment]
Patched with #17184
